SpellCheck/src/main/java/AIR/SpellCheck/Engines/HunspellEngine.java

```diff
@@ -25,44 +27,52 @@
 @Component
 public class HunspellEngine implements ISpellEngine
 {
-  
+  private final static Logger LOG = LoggerFactory.getLogger(HunspellEngine.class);
+
+  private final SpellEngineFactory spellEngineFactory;
+
   @Autowired
-  SpellEngineFactory _spellEngine;
-  
+  public HunspellEngine(final SpellEngineFactory spellEngineFactory) {
+    this.spellEngineFactory = spellEngineFactory;
+  }
+
   @Override
-  public boolean CheckWord (String word) {
+  public boolean CheckWord (final String word) {
+    final String languageCode = getLanguage();
     boolean isCorrect = true;
     try {
-      String languageCode = WebHelper.getQueryString ("lang");
-      if (_spellEngine.getDictionary (languageCode).misspelled(word)) {
+      if (spellEngineFactory.getDictionary (languageCode).misspelled(word)) {
         isCorrect = false;
       }
-      if (_spellEngine.getDictionary (languageCode).misspelled(word)) {
-        isCorrect = false;
-      }
```

Comment:
Slight functionality change.  Previously, the code was checking if the word was misspelled twice.  ...and I have no idea why.  I changed it to only check once.